py4cast/datasets/titan/__init__.py

```diff
@@ -289,6 +282,23 @@ def exist(self, date: dt.datetime, file_format: Literal["npy", "grib"] = "grib")
         return filepath.exists()
 
 
+def process_sample_dataset(date: dt.datetime, params: List[Param]):
+    """Saves each 2D parameter data of the given date as one NPY file."""
+    for param in params:
+        dest_file = param.get_filepath(date, "npy")
+        dest_file.parent.mkdir(exist_ok=True)
+        if not dest_file.exists():
+            try:
+                arr = param.load_data(date, "grib")
+                np.save(dest_file, arr)
+            except Exception as e:
+                print(e)
+                print(
+                    f"WARNING: Could not load grib data {param.name} {param.level} {date}. Skipping sample."
```

Comment:
Isn't it error prone to skip some samples ? Could it lead to uncomplete processed dataset ?

Reply:
I chose to skip the whole sample (1h) when one parameter cannot be loaded because anyway the sample will not be valid and it will not be used during training or test. It's not a problem if we miss a few samples over the whole period. Right now we miss around 500 samples over 20000.